candidatepair.go

```diff
@@ -73,7 +73,7 @@ func (p *candidatePair) Priority() uint32 {
 		return 0
 	}
 
-	return (2^32)*min(g, d) + 2*max(g, d) + cmp(g, d)
+	return (2<<32)*uint64(min(g, d)) + uint64(2*max(g, d)+cmp(g, d))
```

Comment:
Shouldn't this be `1 << 32 `?

Reply:
`2*max(g, d)` overflows if `g == 1<<32-1`

Reply:
yep. fixed.

Reply:
I think there's still an overflow with `2*max(g, d)`